vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/batchRegistration/SampleSpreadsheetLayout.java

```diff
@@ -107,83 +169,303 @@ private static class SampleRegistrationSheetBuilder {
+      conditionsToReplicates = new HashMap<>();
+      for(ExperimentalGroup group : groups) {
+        List<String> varStrings = new ArrayList<>();
+        for(VariableLevel level : group.condition().getVariableLevels()) {
+          String varName = level.variableName().value();
+          String value = level.experimentalValue().value();
+          String unit = "";
+          if(level.experimentalValue().unit().isPresent()) {
+            unit = " "+level.experimentalValue().unit().get();
+          }
+          varStrings.add(varName+":"+value+unit);
+        }
+        String conditionString = String.join("; ", varStrings);
+        conditionsToReplicates.put(conditionString, group.biologicalReplicates());
+      }
+    }
+
+    private List<String> getReplicateLabels() {
+      //TODO values should depend on selected condition!?
+      List<String> replicateLabels = new ArrayList<>();
+      for(List<BiologicalReplicate> replicates : conditionsToReplicates.values()) {
+        replicateLabels.addAll(replicates.stream().map(BiologicalReplicate::label).toList());
+      }
+      return replicateLabels;
+    }
+
+    public void addRow(Spreadsheet spreadsheet) {
+      int rowIndex = findLastRow(spreadsheet)+1;
+
+
+      for(int columnIndex = 0; columnIndex < header.size(); columnIndex++) {
+        String colHeader = header.get(columnIndex);
+        switch (colHeader) {
+          case "Species" -> prefillCell(columnIndex, rowIndex, species, spreadsheet);
+          case "Specimen" -> prefillCell(columnIndex, rowIndex, specimens, spreadsheet);
+          case "Condition" -> prefillCell(columnIndex, rowIndex,
+              conditionsToReplicates.keySet().stream().toList(), spreadsheet);
+          case "Biological replicate id" -> prefillCell(columnIndex, rowIndex, getReplicateLabels(),
+              spreadsheet);
+          default -> {
+            DropDownColumn column = dropdownCellFactory.getColumn(columnIndex);
+            if(column!=null) {
+              column.increaseToRow(rowIndex);
+            }
+          }
+        }
+
+        Cell newCell = spreadsheet.getCell(rowIndex, columnIndex);
+        boolean hasData = !isCellEmpty(newCell);
+        boolean hasDropdown = dropdownCellFactory.findColumnInRange(1, columnIndex)!=null;
+        //cells need to be unlocked if they have no data/dropdown
+        if(!hasData && !hasDropdown) {
+          unlockCell(spreadsheet, rowIndex, columnIndex);
+        }
+      }
+    }
+
+    private int findLastRow(Spreadsheet spreadsheet) {
+      int res = 0;
+      for(int row = 1; row <= Integer.MAX_VALUE; row++) {
+        List<Cell> thisRow = new ArrayList<>();
+        for(int col = 0; col <= header.size(); col++) {
+          thisRow.add(spreadsheet.getCell(row, col));
+        }
+        if (thisRow.stream().allMatch(x -> x == null)) {
+          break;
+        }
+        res = row;
+      }
+      return res;
+    }
```

Comment:
NIT: Do you think it's possible to also define a `lastColumn()` method and scale the spreadsheet accordingly so only actually relevant columns (aka columns with headers) are shown?